### DISCUSSION TITLE
GAIfO doesn't work with Ant-v2. (SOLVED)

### DISCUSSION THREAD

**Discussion by HesNobi** [keiohta/tf2rl#128] [Q&A] [answered]
Hi
I am trying to get GAIfO to run on Ant-v2 with no success, neither by the generated data (TD3) nor the pre-recorded dataset from Berkeley's [D4RL](https://github.com/rail-berkeley/d4rl). However,  it works with HalfCheeta and Pendulum.

I am wondering if you would kindly confirm it.

Thanks for this great project.

**Comment by keiohta** [comment, marked as answer]
Hi @HesNobi I moved your issue to discussion. I'll investigate this later.

**Reply by HesNobi** [reply]
Hi, Would you please put some time on this?
Cause now the GAIL and GAIFO are failing on Ant and Walker2d. I just want to know if it is me or the code.

**Reply by keiohta** [reply]
I'll work on this after finishing current paper deadline. Please wait by then.

**Reply by HesNobi** [reply]
I have found the problem. In order for the SAC to be functional as the generator, it is nessesery to implement the DAC paper. it includes (1) a new reward as R=log(d)-log(1-d) which allows positive and negative values (2) and managing absorbing states. I have opened an issue for this matter. Thanks for the great project.
related to: #141
